ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -307,8 +307,11 @@ public void replicationPauseTest() throws Exception {
   @Test
   public void replicaThreadTest() throws Exception {
     MockClusterMap clusterMap = new MockClusterMap();
-    Host localHost = new Host(clusterMap.getDataNodeIds().get(0), clusterMap);
-    Host remoteHost = new Host(clusterMap.getDataNodeIds().get(1), clusterMap);
+    // to make sure we select hosts with the SPECIAL_PARTITION_CLASS, pick hosts from the replicas of that partition
+    PartitionId specialPartitionId = clusterMap.getWritablePartitionIds(MockClusterMap.SPECIAL_PARTITION_CLASS).get(0);
+    // these hosts have replicas of the "special" partition and all the other partitions.
```

Comment:
I am thinking if we need to check `specialPartitionId == 0`? Considering the `getWritablePartitionIds` depends on `partitionSelectionHelper`, and the latter one relies on `partitions.values()`. Hence, we cannot guarantee `partitions` contains the "special partition".

Reply:
That's true - however, `PartitionSelectionHelper::getPartitionsInClass()` throws an `IllegalArgumentException` if there are no partitions for the class provided - so that should take care of that. Even if that implementation were to change (to say return a null/empty list), since we do a `get(0)` here, there will still be an exception. That should be good enough - what do you think?